ospfd/ospfd.c

```diff
@@ -688,10 +688,11 @@ void ospf_finish(struct ospf *ospf)
 /* Final cleanup of ospf instance */
 static void ospf_finish_final(struct ospf *ospf)
 {
-	struct vrf *vrf;
+	struct vrf *vrf = vrf_lookup_by_id(ospf->vrf_id);
```

Comment:
As you do this lookup for the VRF here, please, change the code at the end of this function to eliminate the double lookup.